shrinkpdf.sh

```diff
@@ -64,16 +64,16 @@ shrink ()
 get_pdf_version() {
   # $1 is the input file. The PDF version is contained in the
   # first 1024 bytes and will be extracted from the PDF file.
-  version=$(head -c 1024 "$1" | grep -Eoa "%PDF-[0-9]\.[0-9]")
+  version=$(cut -b "-1024" "$1" | sed -n '/%PDF-[0-9]\.[0-9]/{s/.*%PDF-/%PDF-/;p;q}' | cut -c "-8")
 
   # If the pdf version could not be extracted, do not change it
   if [ -z "$version" ]; then
-    return
+    return 0
```

Comment:
The function still returns successfully (zero) on errors. The return code should be `1` when an error is encountered.

Reply:
If no version is found, then no version is reported by the function, but then the default will be used.

This way the whole program still works even if this function does not.

Reply:
I now changed it, so that an error is thrown and when the error is caught, the default version is set. I like it a lot better this way.